src/lib/cli_commands/list_steps.rs

```diff
@@ -51,13 +61,17 @@ pub(crate) fn create_list(config: &Config, output_format: &str) -> (String, u32)
         };
 
         if !is_private {
-            count = count + 1;
-
             let category = match task.category {
                 Some(value) => value,
                 None => "No Category".to_string(),
             };
 
+            if category_filter != "default" && category != category_filter {
```

Comment:
we should use an option instead of this. otherwise you can't filter by category = default